### DISCUSSION TITLE
Meeting 2024-10-22

### DISCUSSION THREAD

**Discussion by adamgreig** [rust-embedded/wg#800] [General]
This is the agenda for the next meeting. Please add anything you'd like to discuss below!

* Meetings: Tuesday 8pm Europe/Berlin time
* [Join the Matrix chat](https://matrix.to/#/#rust-embedded:matrix.org)
* [Nominated issues](https://github.com/search?q=org%3Arust-embedded+label%3Anominated+is%3Aopen&type=Issues)
* [IRC/Matrix logs](https://libera.irclog.whitequark.org/rust-embedded/2024-10-22)

## Announcements

* critical-section 1.2 released, notably fixing https://github.com/rust-embedded/critical-section/issues/55
* riscv 0.12 released

**Comment by jamesmunns** [comment]
Alice Ryhl has asked us to review this RFC draft for "target modifiers", or options that could change the ABI that we are compiling for:

https://hackmd.io/f3sFMrYKTWme5vZcUYt1Cw?both

Specifically she has also asked us to help add detail to the "motivation" section for embedded-relevant items

**Comment by jamesmunns** [comment]
Call for Project Goals is [coming this week](https://rust-lang.zulipchat.com/#narrow/channel/392734-council/topic/Project.20Goals.202025H1/near/478330305), time to write any if we have them!

I (personally) think it might be worth proposing goals for:

1. Working on eliding copies, particularly in RVO/NRVO situations, as well as for async task size increases. This came up this week, and likely also fits into other async-forward project goals. CC @Dirbaio 
2. Working on reducing other code-size items, like formatting bloat. CC @diondokter
3. Stabilizing some facets of build-std, which could be used for targets like ESP32 (I think we discussed this?) or other cases where target-features could benefit optimization of stdlib components. CC @MabezDev

**Reply by diondokter** [reply]
I have something that kinda prevents me from joining the meeting lately. So let's just put my thoughts here.

- The std/core/alloc optimize_for_size feature flag is there and anyone can make PRs using it.
- A couple of small low hanging things have been done already
- I don't have any specific suggestion on how to improve sizes further
- There could be big gains in the fmt modules, but someone probably has to dig in deep and into the macros as well
- Whatever the case, we're always limited by the existing public APIs which puts an upper limit on possible gains

**Reply by jannic** [reply]
Existing issue collecting ideas: https://github.com/rust-embedded/wg/issues/775

**Comment by adamgreig** [comment]
@berkus has opened a PR to join the cortex-a team https://github.com/rust-embedded/wg/pull/801

**Comment by adamgreig** [comment]
from last week, cortex-m-pac PR https://github.com/rust-embedded/cortex-m/pull/560#event-14579700710

**Comment by jannic** [comment]
Cortex-m-rt release? https://github.com/rust-embedded/cortex-m/pull/533

**Comment by adamgreig** [comment]
Proposed new svd2rust methods `write_and`, `write_with_zero_and`, `modify_and`: https://github.com/rust-embedded/svd2rust/pull/874

**Comment by berkus** [comment]
aarch64-cpu needs a release, which I will make, following some deps update merges recently.

Additionally, I want to work on providing semi-automatic publishing pipeline (using release-plz probably), to make it more automated. - any objections, comments - lets discuss.
